client/components/details-link/index.tsx

```diff
@@ -3,18 +3,24 @@
 /**
  * External dependencies
  */
-import Gridicon from 'gridicons';
+import * as React from 'react';
+import * as Gridicon from 'gridicons';
 import { addQueryArgs } from '@wordpress/url';
 import { Link } from '@woocommerce/components';
 
-export const getDetailsURL = ( id, parentSegment ) =>
+export const getDetailsURL = ( id: string, parentSegment: string ) =>
 	addQueryArgs( 'admin.php', {
 		page: 'wc-admin',
 		path: `/payments/${ parentSegment }/details`,
 		id,
 	} );
 
-const DetailsLink = ( { id, parentSegment } ) =>
+type DetailsLinkParams = {
+	id?: string;
+	parentSegment: string;
+};
+
+const DetailsLink = ( { id, parentSegment }: DetailsLinkParams ) =>
```

Comment:
With rule @typescript-eslint/explicit-module-boundary-types on, linter complains this exported function is missing return type.

```suggestion
const DetailsLink = ( {
	id,
	parentSegment,
}: DetailsLinkParams ): JSX.Element | null =>
```